components/ratgdo/ratgdo.cpp

```diff
@@ -46,10 +46,23 @@ namespace ratgdo {
 
     void RATGDOComponent::setup()
     {
-        this->pref_ = global_preferences->make_preference<int>(734874333U);
-        if (!this->pref_.load(&this->rollingCodeCounter)) {
+        this->rollingCodePref_ = global_preferences->make_preference<int>(734874333U);
+        if (!this->rollingCodePref_.load(&this->rollingCodeCounter)) {
             this->rollingCodeCounter = 0;
         }
+        this->openingDurationPref_ = global_preferences->make_preference<float>(734874334U);
```

Comment:
Just thinking about what could have caused the crash... Did you ever have a pref with the same type (734874334U) before while testing and it picked up a spurious duration?

Reply:
maybe thats it. 

I've flashed 3 other GDOs and no issues.

I think its fine. I'm going to merge it.

I'll see if I can make it a bit more DRY in a followup without making it less readable.

Reply:
I flashed 7 in total. Only the one I've been doing the development on had the issue so I hope its a fluke